src/containers/Review/assignment/AssignmentTab.tsx

```diff
@@ -103,21 +74,59 @@ const AssignmentTab: React.FC<{
       if (!alreadyAssignedSections.has(section))
         newAssignments[section] = { newAssignee: reviewerId }
     })
+
+    const currentReviewLevel = Math.max(Number(Object.keys(assignmentGroupedLevel)))
     setAssignedSectionsByLevel({ ...assignedSectionsByLevel, [currentReviewLevel]: newAssignments })
   }
 
-  const assignmentGroupedLevel: LevelAssignments = {}
-  assignmentsFiltered.forEach((assignment) => {
-    const { level } = assignment
-    if (!assignmentGroupedLevel[level]) assignmentGroupedLevel[level] = [assignment]
-    else assignmentGroupedLevel[level].push(assignment)
-  })
-
-  const currentReviewLevel = Math.max(Number(Object.keys(assignmentGroupedLevel)))
+  const defaultAssignedSections = Object.values(fullStructure.sections).reduce(
+    (assignedSections, { details: { code } }) => ({
+      ...assignedSections,
+      [code]: { newAssignee: undefined },
+    }),
+    {}
+  )
 
-  const isFullyAssigned = currentReviewLevel
-    ? calculateIsFullyAssigned(assignmentGroupedLevel[currentReviewLevel], sectionCodes)
-    : true
+  const AssignmentSection: React.FC<SectionDetails> = ({ id, title, code }) => {
+    const { reviewStructuresState } = useReviewStructureState()
+    return (
+      <Segment key={id}>
+        <Header className="section-title" as="h5" content={title} />
+        {Object.entries(assignmentGroupedLevel).map(([level, assignments]) => (
+          <div className="flex-column" key={`assignment-group-level-${level}`}>
+            {fullStructure.info.outcome === ApplicationOutcome.Pending && (
+              <AssignmentSectionRow
+                assignments={assignments}
+                sectionCode={code}
+                reviewLevel={Number(level)}
+                structure={fullStructure}
+                assignedSections={assignedSectionsByLevel[level] || defaultAssignedSections}
+                setAssignedSections={(assignedSections) =>
+                  setAssignedSectionsByLevel({
+                    ...assignedSectionsByLevel,
+                    [level]: assignedSections,
+                  })
+                }
+                setEnableSubmit={setEnableSubmit}
+                setAssignmentError={setAssignmentError}
+              />
+            )}
+            {(assignments as AssignmentDetails[]).map((assignment) =>
+              reviewStructuresState[assignment.id] && assignment.isCurrentUserReviewer ? (
+                <ReviewSectionRow
+                  key={`review-row-section-${code}-assignment-${assignment.id}`}
+                  sectionId={id}
+                  fullStructure={fullStructure}
+                  reviewAssignment={assignment}
+                  previousAssignment={assignmentInPreviousStage}
+                />
+              ) : null
+            )}
+          </div>
+        ))}
+      </Segment>
+    )
+  }
```

Comment:
This used to be another component `AssignmentRows` and it was confusing so I thought it would be better to move in here